python/cudf/cudf/_lib/pylibcudf/binaryop.pyx

```diff
@@ -84,3 +85,39 @@ cpdef Column binary_operation(
         raise ValueError(f"Invalid arguments {lhs} and {rhs}")
 
     return Column.from_libcudf(move(result))
+
+
+cpdef bool is_supported_operation(
+    DataType out,
+    DataType lhs,
+    DataType rhs,
+    binary_operator op
+):
+    """Check if an operation is supported for the given data types.
+
+    Parameters
+    ----------
+    out
+        Output data type
+    lhs
+        Left operand type
+    rhs
+        Right operand type
+    op
+        Binary operation to check
+
+    Returns
+    -------
+    True if the operation is supported for the requested types.
+
+    See Also
+    --------
+    :cpp:func:`is_supported_operation`
```

Comment:
FWIW in most of pylibcudf we've been writing this kind of thing not in a separate `See Also` section but more like `For details, see :cpp:func:...` in the main text. Maybe we should migrate to doing this more consistently? I don't recall having a strong reason not. "See Also" is a slightly weaker link than stating explicitly that there's an underlying implementation that this one is meant to reflect "exactly", but that's probably fine.

Reply:
Out of scope for this PR, I'm just asking for future work.